src/googmodule.ts

```diff
@@ -0,0 +1,416 @@
+        const imp = importPathToGoogNamespace(host, sf, importedUrl);
+        modulesManifest.addReferencedModule(sf.fileName, imp.text);
+        const ident: ts.Identifier|undefined = namespaceToModuleVarName.get(imp.text);
+        let initializer: ts.Expression;
+        if (!ident) {
+          namespaceToModuleVarName.set(imp.text, newIdent);
+          initializer = createGoogCall('require', imp);
+        } else {
+          initializer = ident;
+        }
+        const varDecl = ts.createVariableDeclaration(newIdent, /* type */ undefined, initializer);
+        const newStmt = ts.createVariableStatement(
+            /* modifiers */ undefined, ts.createVariableDeclarationList([varDecl]));
+        return ts.setOriginalNode(ts.setTextRange(newStmt, original), original);
+      }
+
+      /**
+       * visitTopLevelStatement implements the main CommonJS to goog.module conversion. It visits a
+       * SourceFile level statement and adds a (possibly) transformed representation of it into
+       * statements. It adds at least one node per statement to statements.
+       *
+       * visitTopLevelStatement:
+       * - converts require() calls to goog.require() calls, with or w/o var assignment
+       * - removes "use strict"; and "Object.defineProperty(__esModule)" statements
+       * - converts module.exports assignments to just exports assignments
+       * - splits __exportStar() calls into require and export (this needs two statements)
+       * - makes sure to only import each namespace exactly once, and use variables later on
+       */
+      function visitTopLevelStatement(
+          statements: ts.Statement[], sf: ts.SourceFile, node: ts.Statement): void {
+        switch (node.kind) {
+          case ts.SyntaxKind.ExpressionStatement: {
+            const exprStmt = node as ts.ExpressionStatement;
+            // Check for "use strict" and certain Object.defineProperty and skip it if necessary.
+            if (isUseStrict(exprStmt) || isEsModuleProperty(exprStmt)) {
+              stmts.push(ts.createNotEmittedStatement(exprStmt));
+              return;
+            }
+            // Check for:
+            //   module.exports = ...;
+            const modExports = rewriteModuleExportsAssignment(exprStmt);
+            if (modExports) {
+              stmts.push(modExports);
+              return;
+            }
+            // Check for:
+            //   "require('foo');" (a require for its side effects)
+            const expr = exprStmt.expression;
+            if (!ts.isCallExpression(expr)) break;
+            let callExpr = expr;
+            // Handle export * in ES5 mode (in ES6 mode, export * is dereferenced already).
+            // export * creates either a pure top-level '__exportStar(require(...))' or the imported
+            // version, 'tslib.__exportStar(require(...))'. The imported version is only substituted
+            // later on though, so appears as a plain "__exportStar" on the top level here.
```

Comment:
Why does the below code look for __export as well?

Reply:
It's `__export()` for the non-tslib version, and `__exportStar` for the tslib case. Fixed the comment.